LFS-707: Improve the look of the quick search results; LFS-264: As a user, I can open the resources provided as suggestions to my quick search in a new browser tab or window

veronikaslc opened the pull request.

Commit:
LFS-707: Improve the look of the quick search results
Added Avatar, removed the created on/created by columns, added subject
type column

Commit:
LFS-707: Improve the look of the quick search results
Fixed the anchor part in the quick search urls path

marta- commented on modules/homepage/src/main/frontend/src/themePage/EntityIdentifier.jsx:
```diff
@@ -48,5 +48,5 @@ export function getEntityIdentifierLink(row) {
       path = `/content.html/admin${row["@path"]}`;
     }
-    return (<Link to={path}>{id}</Link>);
+    return (<a href={path}>{id}</a>);
```
This `<a>` in combination with the routing code that runs on the `click` event of each `MenuItem` appears to have the following effect:
* when clicking on an item for a _form result_ outside of the `<a>` element, the result is displayed with the right question highlighted (correct)
* when clicking on the `<a>` element directly, the result is displayed briefly with the right question highlighted, then the page is reloaded and the question is no longer highlighted (incorrect)
* when opening the link in a new tab, the result is displayed, but without the proper highlight (better than before, but still incomplete wrt https://phenotips.atlassian.net/browse/LFS-264).

I suggest to remove this change from the current PR and do it as part of a separate one, for LFS-264, in combination with adjustments to the `SearchBar`'s `MenuItem onClick` code.

marta- commented on modules/homepage/src/main/frontend/src/themePage/Navbars/HeaderSearchBar.jsx:
```diff
@@ -20,68 +20,33 @@ import PropTypes from "prop-types";
-      />
-    </React.Fragment>
+              <ListItemAvatar>
+                {MatchAvatar(resultData, classes)}
+              </ListItemAvatar>
+              <ListItemText
+                primary={getEntityIdentifierLink(resultData)}
+                secondary={QuickSearchMatch(resultData[LFS_QUERY_MATCH_KEY], classes)}
```
`MatchAvatar` and `QuickSearchMatch` should be called as actual components, not simple functions.

marta- reviewed (changes requested):
The dropdown would look more polished if the avatar was at the top of the item, instead of being at the center.

Current look:
![image (13)](https://user-images.githubusercontent.com/651980/99320756-4ec9e300-283a-11eb-8486-b8ba8687520b.png)


Suggested look:
![image (14)](https://user-images.githubusercontent.com/651980/99320763-54272d80-283a-11eb-858c-d1f9528576fe.png)


Can be achieved by adding `alignItems="flex-start"` to each result `MenuItem` in `SearchBar.jsx`.

Commit:
LFS-707: Improve the look of the quick search results
Refactored code to aggregate the common functionality in QuickSearchIdentifier component

veronikaslc commented:
Codereview comments addressed. Last commit should also fix 
[LFS-264](https://phenotips.atlassian.net/browse/LFS-264) As a user, I can open the resources provided as suggestions to my quick search in a new browser tab or window

marta- reviewed (changes requested):
Looks great and opening match links now behaves as expected.

There's only one small ux issue now: when clicking on the match entry outside of the link, the entry ripples and the dropdown closes, but the match isn't displayed.

![captured (10)](https://user-images.githubusercontent.com/651980/99409305-38b23600-28bf-11eb-9be0-465651a5627d.gif)

Either the menu entry should not react and the menu should no close, or the click should be  somehow forwarded to the link. What happens right now is confusing.

marta- commented on modules/homepage/src/main/frontend/src/themePage/Navbars/HeaderSearchBar.jsx:
```diff
@@ -20,81 +20,25 @@ import PropTypes from "prop-types";
 
 HeaderSearchBar.propTypes = {
-  invertColors: PropTypes.bool,
-  doNotEscapeQuery: PropTypes.bool
```
I didn't look in too much detail yet, but are we sure it's ok to completely get rid of this?

veronikaslc replied:
It's not used anywhere so far.

marta- replied:
The Query builder still needs it, It should be put back: https://github.com/ccmbioinfo/lfs/blob/2b0a98c3116cfda676024ae33a59d7faf1c28e4c/modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/QueryBuilder.java#L181

marta- commented:
> Codereview comments addressed. Last commit should also fix
> [LFS-264](https://phenotips.atlassian.net/browse/LFS-264) As a user, I can open the resources provided as suggestions to my quick search in a new browser tab or window

It would be useful to tag the commits that fix LFS-264 with the issue id so we have a trace of it in the revision history for future reference. Alternatively we can squash the commits upon merge and make sure both 707 and 264 are referenced in that commit.

marta- commented on modules/homepage/src/main/frontend/src/themePage/Navbars/QuickSearchIdentifier.jsx:
```diff
@@ -0,0 +1,97 @@
+    if (resultData["jcr:primaryType"] == "lfs:Questionnaire") {
+      fullPath = `/content.html/admin${resultData["@path"]}#${anchorPath}`;
+    }
+    return (<ListItem className={classes.quickSearchResultsIdentifier}>
```
We really don't need a class just to specify `alignItems`.
```suggestion
    return (<ListItem alignItems="flex-start">
```

Commit:
LFS-707: Improve the look of the quick search results
Rendering ListItem as a link

Commit:
LFS-707: Improve the look of the quick search results
Simplified styling

marta- commented on modules/commons/src/main/frontend/src/headerStyle.jsx:
```diff
@@ -130,7 +130,11 @@ const headerStyle = theme => ({
-  }
+  },
+  quicksearchIdentifierLink: {
+    color: infoColor,
```
`infoColor` appears to actually be a set of 4 colors, causing the link to remain colored black.
![image](https://user-images.githubusercontent.com/651980/99429378-e7fa0780-28d5-11eb-90d9-f4a9e18bf568.png)

I actually like it more without the blue color and underline, so I wouldn't mind removing the `quicksearchIdentifierLink` completely.

marta- reviewed (changes requested):
Works great. Now only to remove the match link styling and put back the `doNotEscape` property.

Commit:
LFS-707: Improve the look of the quick search results
Removed link id styling; Added back doNotEscapeQuery param as it is neeed in the Query builder

marta- reviewed (approved):
:tada:

sashaandjic commented:
tested; all good
* I noticed that the order of the fields is different, however I'm not sure if this has anything to do with the PR
![image](https://user-images.githubusercontent.com/23243244/99571847-d716c800-29a1-11eb-99f8-8288b9cdd9fe.png)
![image](https://user-images.githubusercontent.com/23243244/99571866-db42e580-29a1-11eb-9d12-9aa244aad5d9.png)
